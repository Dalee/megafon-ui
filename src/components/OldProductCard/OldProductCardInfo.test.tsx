import * as React from 'react';
import { shallow } from 'enzyme';
import OldProductCardInfo from './OldProductCardInfo';
import Checked from 'icons/checked_24.svg';

const props = {
    title: 'Включайся!',
    link: 'https://www.google.com/',
    description: 'Амедиатека',
    descriptionIcon: <Checked style={{ width: '40px', height: '40px' }} />,
    additionalParams: [
        {
            value: '300',
            unit: 'минут',
            title: 'Звонки на все номера России',
        }, {
            value: '300',
            unit: 'SMS',
            title: 'SMS на номера Домашнего региона',
        }, {
            value: '4',
            unit: 'ГБ',
            title: 'На любые сервисы',
        },
    ],
    badges: [{
        title: 'title',
        code: 'code',
        hint: 'hint',
    },
    {
        title: 'title',
        code: 'code',
    }],
};

describe('<OldProductCardInfo />', () => {
    it('it renders OldProductCardInfo', () => {
        const wrapper = shallow(<OldProductCardInfo {...props}><div /></OldProductCardInfo>);
        expect(wrapper).toMatchSnapshot();
    });
});
