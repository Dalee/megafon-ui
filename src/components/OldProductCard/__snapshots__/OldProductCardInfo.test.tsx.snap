// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OldProductCardInfo /> it renders OldProductCardInfo 1`] = `
<div
  className="old-product-card-info"
>
  <div
    className="old-product-card-info__description"
  >
    <Header
      as="h2"
      className="old-product-card-info__description-title"
      color="black"
      margin={false}
    >
      Включайся!
    </Header>
    <div
      className="old-product-card-info__badges"
    >
      <div
        className="old-product-card-info__badge old-product-card-info__badge_type_code"
        key="title0"
      >
        <BubbleHint
          placement="right"
          popupAlign="left"
          popupPadding="normal"
          popupWidth="small"
          trigger={
            <div
              className="old-product-card-info__badge-title"
            >
              title
            </div>
          }
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "hint",
              }
            }
          />
        </BubbleHint>
      </div>
      <div
        className="old-product-card-info__badge old-product-card-info__badge_type_code"
        key="title1"
      >
        <div
          className="old-product-card-info__badge-title"
        >
          title
        </div>
      </div>
    </div>
    <div
      className="old-product-card-info__description-wrap"
    >
      <test-file-stub
        className="old-product-card-info__description-icon"
        style={
          Object {
            "height": "40px",
            "width": "40px",
          }
        }
      />
      <div
        className="old-product-card-info__description-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Амедиатека",
          }
        }
      />
    </div>
    <TextLink
      className="old-product-card-info__description-more"
      color="blue"
      href="https://www.google.com/"
      onClick={[Function]}
      target="_blank"
      underlineStyle="solid"
      underlineVisibility="hover"
    >
      Подробнее
    </TextLink>
  </div>
  <div
    className="old-product-card-info__params"
  >
    <ul
      className="old-product-card-info__params-list"
    >
      <li
        className="old-product-card-info__params-item"
        key="Звонки на все номера России0"
      >
        <span
          data-amount="300"
        >
          300
        </span>
         
        минут
      </li>
      <li
        className="old-product-card-info__params-item"
        key="SMS на номера Домашнего региона1"
      >
        <span
          data-amount="300"
        >
          300
        </span>
         
        SMS
      </li>
      <li
        className="old-product-card-info__params-item"
        key="На любые сервисы2"
      >
        <span
          data-amount="4"
        >
          4
        </span>
         
        ГБ
      </li>
    </ul>
  </div>
</div>
`;
